modules/openapi-generator/src/main/resources/python/python-experimental/api_client.mustache

```diff
@@ -542,3 +566,159 @@ class ApiClient(object):
+        if resource_path is None:
+            raise Exception("Resource path must be set")
+        if body is None:
+            body = ''
+        else:
+            body = json.dumps(body)
+
+        target_host = urlparse(self.configuration.host).netloc
+        target_path = urlparse(self.configuration.host).path
+
+        request_target = method.lower() + " " + target_path + resource_path
+
+        if query_params:
+            raw_query = urlencode(query_params).replace('+', '%20')
+            request_target += "?" + raw_query
+
+        # Get current time and generate RFC 1123 (HTTP/1.1) date/time string.
+        now = datetime.datetime.now()
+        stamp = mktime(now.timetuple())
+        cdate = formatdate(timeval=stamp, localtime=False, usegmt=True)
+        created = now.strftime("%s")
+
+        signed_headers = {}
+        signed_header_dict = {}
+        for hdr_key in self.configuration.signed_headers:
+            hdr_key = hdr_key.lower()
+            if hdr_key == '(request-target)':
+                value = request_target
+            elif hdr_key == '(created)':
+                value = created
+            elif hdr_key == 'date':
+                value = cdate
+                signed_header_dict['Date'] = '{0}'.format(cdate)
+            elif hdr_key == 'digest':
+                request_body = body.encode()
+                body_digest, digest_prefix = self.get_message_digest(request_body)
+                b64_body_digest = b64encode(body_digest.digest())
+                value = digest_prefix + b64_body_digest.decode('ascii')
+                signed_header_dict['Digest'] = '{0}{1}'.format(digest_prefix, b64_body_digest.decode('ascii'))
+            elif hdr_key == 'host':
+                value = target_host
+                signed_header_dict['Host'] = '{0}'.format(target_host)
+            else:
+                value = headers[hdr_key]
+            signed_headers[hdr_key] = value
+
+        if len(self.configuration.signed_headers) == 0:
+            signed_headers['(created)'] = created
+
+        string_to_sign = ""
+        for i, (key, value) in enumerate(signed_headers.items()):
+            string_to_sign += "{0}: {1}".format(key.lower(), value)
+            if i > 0:
+                string_to_sign += "\n"
+
+        digest, digest_prefix = self.get_message_digest(string_to_sign.encode())
+        b64_signed_msg = self.sign_digest(digest)
+
+        auth_header = self.get_authorization_header(signed_headers, b64_signed_msg, created)
+        signed_header_dict['Authorization'] = '{0}'.format(auth_header)
+
+        return signed_header_dict
+
+    def get_message_digest(self, data):
+        """
+        Calculates and returns a cryptographic digest of a specified HTTP request.
+
+        :param data: The string representation of the date to be hashed with a cryptographic hash.
+        :return: The hashing object that contains the cryptographic digest of the HTTP request. 
```

Comment:
This returns a tuple of (digest, prefix) can you describe both returned params and types in this docstring?

Reply:
updated documentation.